homeassistant/components/slack/notify.py

```diff
@@ -15,157 +16,141 @@
-                    # Prepare structure for Slack API
-                    data = {
-                        "content": None,
-                        "filetype": None,
-                        "filename": filename,
-                        # If optional title is none use the filename
-                        "title": title if title else filename,
-                        "initial_comment": message,
-                        "channels": target,
-                    }
-                    # Post to slack
-                    self.slack.files.post(
-                        "files.upload", data=data, files={"file": file_as_bytes}
-                    )
-                else:
-                    self.slack.chat.post_message(
-                        target,
-                        message,
-                        as_user=self._as_user,
-                        username=self._username,
-                        icon_emoji=self._icon,
-                        attachments=attachments,
-                        link_names=True,
-                    )
-            except slacker.Error as err:
-                _LOGGER.error("Could not send notification. Error: %s", err)
-
-    def load_file(
-        self, url=None, local_path=None, username=None, password=None, auth=None
-    ):
-        """Load image/document/etc from a local path or URL."""
-        try:
-            if url:
-                # Check whether authentication parameters are provided
-                if username:
-                    # Use digest or basic authentication
-                    if ATTR_FILE_AUTH_DIGEST == auth:
-                        auth_ = HTTPDigestAuth(username, password)
-                    else:
-                        auth_ = HTTPBasicAuth(username, password)
-                    # Load file from URL with authentication
-                    req = requests.get(url, auth=auth_, timeout=CONF_TIMEOUT)
-                else:
-                    # Load file from URL without authentication
-                    req = requests.get(url, timeout=CONF_TIMEOUT)
-                return req.content
-
-            if local_path:
-                # Check whether path is whitelisted in configuration.yaml
-                if self.is_allowed_path(local_path):
-                    return open(local_path, "rb")
-                _LOGGER.warning("'%s' is not secure to load data from!", local_path)
-            else:
-                _LOGGER.warning("Neither URL nor local path found in parameters!")
-
-        except OSError as error:
-            _LOGGER.error("Can't load from URL or local path: %s", error)
-
-        return None
+        if ATTR_FILE in data:
+            return await self._async_send_local_file_message(
+                data[ATTR_FILE], targets, message, title
+            )
+
+        attachments = data.get(ATTR_ATTACHMENTS, {})
+        if attachments:
+            _LOGGER.warning(
+                "Attachments are now part of Slack's legacy API; "
```

Comment:
Do we need to keep support for attachments? Can we just make attachments use blocks instead and hide the api details from the user?

Reply:
Unfortunately, the underlying data models are pretty different. That said, since I've already introduced a breaking change and this is a feature that Slack won't build upon long term, I don't have a problem removing it outright. What do you think @MartinHjelmare?

Reply:
I don't know how much this feature is used. Maybe it needs a deprecation period first?

Reply:
I love a good deprecation. 😄 I already have one planned for 0.114-ish; I'll do the same here. Thanks!

Reply:
@MartinHjelmare Is updating the warning with deprecation language (a la fcaedd7) sufficient?

Reply:
Sounds good!